test/headers/mock_usart.hpp

```diff
@@ -0,0 +1,54 @@
+#pragma once
+#include <usart_connection.hpp>
+#include <vector>
+
+namespace r2d2 {
+
+    class mock_usart_c : public usart_connection_c {
```

Comment:
Whats a mock usart doing in your feature, it is terribly misplaced and should probaply be moved to the usart library

Reply:
Since this module has be transferd i think it is best the new team(which im a part off) change the implementation to the new library implementation, since it works at this point in time we might need to merge it as is and change it as soon as possible when this is implemented in the library